pages/deep_well_plates.vue

```diff
@@ -0,0 +1,168 @@
+<template>
+  <b-container>
+    <h1 class="mt-3 mb-3">Deep-Well Plates</h1>
+    <b-card
+      class="input-area"
+      title="Import Deep-Well Plates"
+      subtitle="Add externally sourced deep-well plates to Sequencescape so that they can be stamped to shallow-well plates using Limber."
+    >
+      <label for="plate-barcodes" class="mt-3 mb-2">Please scan plate barcode(s):</label>
+      <b-form-textarea
+        id="plate-barcodes"
+        ref="plateBarcodesInput"
+        v-model="plateBarcodes"
+        rows="5"
+        max-rows="6"
+        @keydown.ctrl.enter="submitBarcodes()"
+      />
+      <p class="text-muted text-end tip"><b>Tip:</b> You can also submit with Ctrl + Enter.</p>
+      <b-button
+        id="clearBarcodes"
+        variant="outline-secondary"
+        class="me-2"
+        @click="clearBarcodes()"
+      >
+        Clear
+      </b-button>
+      <div class="float-end">
+        <b-button
+          id="submitBarcodes"
+          variant="primary"
+          :disabled="isBusy() || !isValid()"
+          @click="submitBarcodes()"
+        >
+          Submit
+          <b-spinner v-show="isBusy()" id="busySpinner" small></b-spinner>
+        </b-button>
+      </div>
+    </b-card>
+    <br />
+    <h3>Import Results</h3>
+    <b-table
+      id="resultsTable"
+      class="mt-3"
+      show-empty
+      responsive
+      :items="results"
+      :fields="resultFields"
+      hover
+      head-variant="light"
+    />
+  </b-container>
+</template>
+
+<script lang="ts">
```

Comment:
Ah this explains the use of typescript linter. Im all for typescript but just want to make sure its the planned direction for the entire codebase? Maybe a readme comment saying to use typescript for all new components or something?

Reply:
Yeah, you don't actually get a choice if you want to adopt Nuxt 3 properly.  Notice that this is not defined like a past component/page.  It is using `export default defineComponent()` instead of just `export default {}`.  This kicks in typescript for you and is the way components are supposed to be defined in the Nuxt 3 world.  If you don't include the declaration of `lang="ts"` then you get a linting error about using JavaScript without explicitly allowing it.  I didn't feel it was right to adopt Nuxt 3 and then try to force old conventions on it for a new component/page.  So there's no need to add it to the Readme.  The linter will stop you from doing it wrong.

![image](https://user-images.githubusercontent.com/1999241/233985848-0c6c0ca8-951f-4535-9efb-17187dc4863c.png)

Reply:
Ah ok cool. Can you use the new vue3 options API and avoid that entirely by defining the entire script as a setup?

Reply:
Honestly I'm not sure.  I haven't used any `<script setup>` blocks yet, but having read a bit about what the setup script does, I think it's not suitable for declaring your component.  The standard `<script>` block is used to declare the component and it run once only while the `<script setup>` block is run every time you create an instance of the component and provides a way of interacting with the initial props.  You use both.